src/SerilogWeb.Classic.Mvc/SerilogWebClassicMvcLoggerConfigurationExtensions.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Serilog.Configuration;
+using SerilogWeb.Classic.Mvc.Enrichers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Serilog
+{
+    /// <summary>
+    /// Extends <see cref="LoggerConfiguration"/> to add enrichers for SerilogWeb.Classic.Mvc logging module 
+    /// </summary>
+    public static class SerilogWebClassicMvcLoggerConfigurationExtensions
+    {
+        /// <summary>
+        /// Enrich log events with the name of the current MVC action.
+        /// </summary>
+        /// <param name="enrichmentConfiguration">Logger enrichment configuration.</param>
+        /// <returns>Configuration object allowing method chaining.</returns>
+        public static LoggerConfiguration WithMvcActionName(
```

Comment:
It would probably look nicer to use parameters with default values instead here, and have only one extension method 

`WithMvcActionName(this LoggerEnrichementConfiguration enrichmentConfiguration,
                   string propertyName="MvcAction")`


- This will work for AppSettings (it usesdefault values of parameters when they are not provided)
- that will make this file a bit shorter
- the default value will show up in Intellisense when using the extension method from code ... 

Maybe it would make sense to also refactor the constructors of the Enricher so that there is only one constructor with optional parameters instead of several constructors.

(This comment applies to all 4 `MvcActionNameEnricher`, `MvcControllerNameEnricher`, `MvcRouteDataEnricher`, ``MvcRouteTemplateEnricher`)

What do you think ?

Reply:
I separated the methods mostly to avoid duplicating the default property names in the codebase, but maybe we could introduce constants to avoid this problem:

```csharp
    public class MvcActionNameEnricher : BaseMvcContextInfoEnricher
    {
        public const string MvcActionPropertyName = "MvcAction";

        public MvcActionNameEnricher()
            : this(MvcActionPropertyName)
        {
        }
        // (...)
    }

    public static class SerilogWebClassicMvcLoggerConfigurationExtensions
    {
        public static LoggerConfiguration WithMvcActionName(
            this LoggerEnrichmentConfiguration enrichmentConfiguration,
            string propertyName = MvcActionNameEnricher.MvcActionPropertyName)
        {
            if (enrichmentConfiguration == null) throw new ArgumentNullException(nameof(enrichmentConfiguration));
            return enrichmentConfiguration.With(new MvcActionNameEnricher(propertyName));
        }
        // (...)
    }
```

Reply:
I dont think we should refactor the constructors to leave only one, because that would make illegal this code:

```csharp
new LoggerConfiguration()
    .Enrich.With<MvcActionNameEnricher>();
```

Reply:
Yes, putting the default property names as constant sounds like a clean way to do it 👍 

About optional parameters in the constructors ... 🤔 You are right, in my head, under the hood, having a constructor with only optional parameters is equivalent to having a constructor with no parameters (aka default constructor) and other constructors with *non optional* parameters ... 

but it turns out it's not quite the same : https://stackoverflow.com/questions/23390426/constructor-with-optional-parameter-violates-new-constraint?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa :) 

Let's leave the constructors alone, then :) 

Thank you !